ggshield/core/git_shell.py

```diff
@@ -280,20 +280,21 @@ def read_git_file(ref: str, path: Path, wd: Optional[str] = None) -> str:
 def tar_from_ref_and_filepaths(
     ref: str,
     filepaths: Iterable[Path],
-    acceptation_func: Optional[Callable[[Path, Callable[[], str]], bool]] = None,
+    acceptation_func: Optional[Callable[[Path, str], bool]] = None,
     wd: Optional[str] = None,
 ) -> bytes:
     """
     Builds a gzipped archive from a given git reference, and selected filepaths.
     The filepaths are typically obtained via `get_filepaths_from_ref` or `get_staged_filepaths`
     before being filtered.
     The archive is returned as raw bytes.
+    For better performance, pre-filter the provided filepaths, or use the `acceptation_func`
```

Comment:
Filtering based on content could also be done before using this function. IMO it should be a function which generate a tar from a list of Path.